l10n_fr_payment_payfip/models/payment_transaction.py

```diff
@@ -0,0 +1,159 @@
+            ('V', "Effective payment (V)"),
+            ('A', "Abandoned payment (A)"),
+            ('R', "Other cases (R)"),
+            ('Z', "Other cases (Z)"),
+            ('U', "Unknown"),
+        ]
+    )
+
+    payfip_amount = fields.Float(
+        string="PayFIP amount",
+    )
+
+    # endregion
+
+    def create(self, vals):
+        res = super(PayFIPTransaction, self).create(vals)
+        if res.acquirer_id.provider == 'payfip':
+            prec = self.env['decimal.precision'].precision_get('Product Price')
+            email = res.partner_email
+            amount = int(float_round(res.amount * 100.0, prec))
+            reference = res.reference.replace('-', ' ')
+            acquirer_reference = '%.15d' % int(
+                uuid.uuid4().int % 899999999999999)
+            res.acquirer_reference = acquirer_reference
+            idop = res.acquirer_id.payfip_get_id_op_from_web_service(
+                email, amount, reference, acquirer_reference)
+            res.payfip_operation_identifier = idop
+
+        return res
+
+    def _get_specific_rendering_values(self, processing_values):
+        res = super()._get_specific_rendering_values(processing_values)
+        if self.provider != 'payfip':
+            return res
+
+        base_url = self.acquirer_id.get_base_url()
+        return {
+            'api_url': PayFIPController._payment_url,
+            'numcli': self.acquirer_id.payfip_customer_number,
+            'exer': fields.Datetime.now().year,
+            'refdet': self.acquirer_reference,
+            'objet': self.reference,
+            'montant': self.amount,
+            'mel': self.partner_email,
+            'urlnotif': urls.url_join(base_url, PayFIPController._notification_url),
+            'urlredirect': urls.url_join(base_url, PayFIPController._return_url),
+            'saisie': 'W' if self.acquirer_id.state == 'prod' else 'T',
+        }
+
+    @api.model
+    def _get_tx_from_feedback_data(self, provider, data):
+        """ Override of payment to find the transaction based on Payfip data.
+
+        param str provider: The provider of the acquirer that handled the transaction
+        :param dict data: The feedback data sent by the provider
+        :return: The transaction if found
+        :rtype: recordset of `payment.transaction`
+        :raise: ValidationError if the data match no transaction
+        """
+        tx = super()._get_tx_from_feedback_data(provider, data)
+        if provider != 'payfip':
+            return tx
+
+        reference = data["objet"]
+        tx = self.sudo().search(
+            [('reference', '=', reference), ('provider', '=', 'payfip')])
+        if not tx:
+            raise ValidationError(
+                "PayFIP: " +
+                _("No transaction found matching reference %s.", reference)
+            )
+        return tx
+
+    def _process_feedback_data(self, feedback_data):
+        data = self.acquirer_id.payfip_get_result_from_web_service(
```

Comment:
I supposed you should call super to avoid breakng inheritance in case of multiple provider installed on the same databse

```suggestion
       super()._process_feedback_data(feedback_data)
        if self.provider != "payfip":
            return
        data = self.acquirer_id.payfip_get_result_from_web_service(
```